lib/experimental/class-wp-html-walker-test.php

```diff
@@ -0,0 +1,192 @@
+	}
+
+	public function test_finding_existing_tag_returns_the_walker_object() {
+		$w      = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$result = $w->find_next_tag();
+		$this->assertSame( $w, $result, 'Finding an existing tag returns the Walker object' );
+	}
+
+	public function test_finding_non_existing_tag_returns_false() {
+		$w      = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$result = $w->find_next_tag( 'p' );
+		$this->assertFalse( $result );
+	}
+
+	public function test_set_new_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->set_attribute( 'test-attribute', 'test-value' );
+		$this->assertSame( '<div test-attribute="test-value" id="first"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_set_existing_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->set_attribute( 'id', 'new-id' );
+		$this->assertSame( '<div id="new-id"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_remove_existing_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->remove_attribute( 'id' );
+		$this->assertSame( '<div ><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_remove_non_existing_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->remove_attribute( 'no-such-attribute' );
+		$this->assertSame( '<div id="first"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_add_class_when_there_is_no_class_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->add_class( 'foo-class' );
+		$this->assertSame( '<div class="foo-class" id="first"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_add_two_classes_when_there_is_no_class_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->add_class( 'foo-class' )->add_class( 'bar-class' );
+		$this->assertSame( '<div class="foo-class bar-class" id="first"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_remove_class_when_there_is_no_class_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$w->find_next_tag()->remove_class( 'foo-class' );
+		$this->assertSame( '<div id="first"><span id="second">Text</span></div>', $w . '' );
+	}
+
+	public function test_add_class_when_there_is_a_class_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_WITH_CLASSES );
+		$w->find_next_tag()->add_class( 'foo-class' )->add_class( 'bar-class' );
+		$this->assertSame( '<div class="main with-border foo-class bar-class" id="first"><span class="not-main bold with-border" id="second">Text</span></div>',
+			$w . '' );
+	}
+
+	public function test_remove_class_when_there_is_a_class_attribute() {
+		$w = new WP_HTML_Walker( self::HTML_WITH_CLASSES );
+		$w->find_next_tag()->add_class( 'foo-class' )->add_class( 'bar-class' );
```

Comment:
This test is copied from the one above and needs some adjustments.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/42485/commits/5bf0c2815785c2a7f91fe844aa82cc0cafb3f5af